api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -171,6 +171,21 @@ message HttpConnectionManager {
   // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
   uint32 xff_num_trusted_hops = 19;
 
+  // If set, Envoy will not append the remote address to the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP header. This may be used in
+  // conjunction with HTTP filters that explicitly manipulate XFF after the HTTP connection manager
+  // has mutated the request headers.
+  bool skip_xff_append = 21;
```

Comment:
How is this different from setting "use_remote_address" to false? https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_utility.cc#L79. I think maybe we don't need this option and you only need to disable the internal/external logic?

Reply:
If you set `use_remote_address` to false, then Envoy trusts the incoming XFF when determining remote address for logging, which we don't want hear. We want to avoid adding new XFF, while having Envoy still look at the connection for determining the remote address for non-XFF purposes.